client/blocks/importer/hooks/use-site-can-migrate.ts

```diff
@@ -35,29 +35,34 @@ export function useSiteMigrateInfo(
 		);
 	};
 
-	const onMigrationEnabledSuccess = ( data: MigrationEnabledResponse ) => {
-		const canMigrate = checkCanSiteMigrate( data );
-		if ( canMigrate ) {
-			setSiteCanMigrate( true );
-			setSourceSiteId( data.source_blog_id );
-		} else {
-			resetMigrationEnabled();
+	const {
+		refetch,
+		isFetching: isMigrationEnabledFetching,
+		isError,
+		isSuccess,
+		data,
+	} = useMigrationEnabledInfoQuery( targetSiteId, sourceSiteSlug, fetchMigrationEnabledOnMount );
+
+	useEffect( () => {
+		if ( isSuccess && data ) {
```

Comment:
Is it possible that the request will be successful but here will be no `data`? Previously we were not checking the `data` at all. Or is it to satisfy TS?

Reply:
Yeah, I think there might be cases that's success without data, we did do a check in the `checkCanSiteMigrate` function. So we can remove the data check here.

Reply:
Agreed. We don't need the `data` check. There won't be a `success` status without `data`.